### DISCUSSION TITLE
Aiuto per Automazione accensione luci

### DISCUSSION THREAD

**Discussion by Moxser** [indomus/forum#1258] [HUB: Home Assistant]
Ciao a tutti e Buon Anno a tutti.
Da diverso tempo utilizzo un automazione che mi gestisce le luci quando rientro a casa.
Ora vorrei migliorarla perché con il tempo ho capito che ha delle lacune ma non so proprio come fare.

E' molto semplice quando il "Gruppo Famiglia" (composto dal device tracker mio e della mia compagna) arriva a casa PER 1 minuto e mezzo e il sole è sotto l'orizzonte accende una scena da me impostata. Il problema è che se io passo davanti a casa ma poi mi allontano (abito in un condominio e ho la strada principale proprio davanti) HA in ogni caso dopo 1 minuto e mezzo mi accende la scena per poi spegnerla con un altra automazione creata per spegnere tutto quando il "Gruppo Famiglia" non è presente.
Io vorrei se possibile accendere l'automazione SOLO se il gruppo famiglia è a casa per un determinato tempo. Quindi una sorta di conto alla rovescia che se non si conclude non attiva nulla.
Al posto del "FOR" posso usare qualcos'altro?

Qui sotto vi allego l'automazione:

``` alias: Accensione Luce rientro
description: >-
  Attivazione della scena rientro a casa quando torniamo e solo se il sole è
  sotto l'orizzonte
trigger:
  - platform: state
    entity_id: group.famiglia
    to: home
    from: not_home
    for:
      hours: 0
      minutes: 1
      seconds: 30
condition:
  - condition: state
    entity_id: sun.sun
    state: below_horizon
action:
  - scene: scene.ritorno_a_casa
  - type: turn_off
    device_id: eb77e37fafe7aba217c9326f8426457e
    entity_id: light.color_temperature_light_1
    domain: light
  - service: notify.mobile_app_enrico
    data:
      message: Luci rientro accese
      title: Illuminazione zona giorno
mode: single
```
Grazie mille per l'aiuto

**Comment by snakuzzo** [comment]
Il `for` è ok...e fa proprio quello di cui hai bisogno. E' di fatto un timer che parte al cambio stato del tracker ad `home` e si interrompe se cambia a `not_home`. 
Il tuo problema è che probabilmente lo stato dei tracker (tocca vedere quale) non cambia quando ti allontani dopo esserti avvicinato. Verifica questo innanzi tutto. Considera che l'aggiornamento delle posizioni dei tracker è un tema abbastanza delicato e può cambiare da device a device...non è proprio il massimo della precisione.

**Reply by Moxser** [reply]
Certo capisco.
Forse l'unica cosa che posso fare è aumentare il tempo per capire se vado a bypassare il problema. 
Grazie mille Diego
